src/main/osd/osd_elements.c

```diff
@@ -2094,81 +2168,131 @@ uint8_t osdGetActiveElementCount(void)
-        } else {
-            len = tfp_sprintf(buff, "%s", "RPM LIMIT OFF");
+        {
+            const bool rpmLimitActive = mixerConfig()->rpm_limit > 0 && isMotorProtocolBidirDshot();
+            if (rpmLimitActive) {
+                len = tfp_sprintf(buff, "RPM LIMIT ON  %d", mixerConfig()->rpm_limit_value);
+            } else {
+                len = tfp_sprintf(buff, "%s", "RPM LIMIT OFF");
+            }
+            displayWrite(osdDisplayPort, midCol - len/2, currentRow++, DISPLAYPORT_SEVERITY_NORMAL, buff);
+
+            if (rpmLimitActive) {
+                specState = POLES;
+            } else {
+                specState = THR;
+            }
         }
+        break;
+
+    case POLES:
+        memset(buff,0,strlen(buff));
+        len = tfp_sprintf(buff, "KV %d   POLES %d", motorConfig()->kv, motorConfig()->motorPoleCount);
         displayWrite(osdDisplayPort, midCol - len/2, currentRow++, DISPLAYPORT_SEVERITY_NORMAL, buff);
 
-        if (rpmLimitActive) {
-            memset(buff,0,strlen(buff));
-            len = tfp_sprintf(buff, "KV %d   POLES %d", motorConfig()->kv, motorConfig()->motorPoleCount);
-            displayWrite(osdDisplayPort, midCol - len/2, currentRow++, DISPLAYPORT_SEVERITY_NORMAL, buff);
+        specState = MIXER;
+        break;
 
-            memset(buff,0,strlen(buff));
-            len = tfp_sprintf(buff, "%d  %d  %d", mixerConfig()->rpm_limit_p, mixerConfig()->rpm_limit_i, mixerConfig()->rpm_limit_d);
-            displayWrite(osdDisplayPort, midCol - len/2, currentRow++, DISPLAYPORT_SEVERITY_NORMAL, buff);
-        } else
+    case MIXER:
+        memset(buff,0,strlen(buff));
+        len = tfp_sprintf(buff, "%d  %d  %d", mixerConfig()->rpm_limit_p, mixerConfig()->rpm_limit_i, mixerConfig()->rpm_limit_d);
+        displayWrite(osdDisplayPort, midCol - len/2, currentRow++, DISPLAYPORT_SEVERITY_NORMAL, buff);
+
+        specState = THR;
+        break;
+
+    case THR:
 #endif // #USE_RPM_LIMIT
-        {
-            memset(buff,0,strlen(buff));
-            len = tfp_sprintf(buff, "THR LIMIT %s", lookupTableThrottleLimitType[currentControlRateProfile->throttle_limit_type]);
-            if (currentControlRateProfile->throttle_limit_type != THROTTLE_LIMIT_TYPE_OFF) {
-                len = tfp_sprintf(buff, "%s %d", buff, currentControlRateProfile->throttle_limit_percent);
-            }
-            displayWrite(osdDisplayPort, midCol - len/2, currentRow++, DISPLAYPORT_SEVERITY_NORMAL, buff);
+        memset(buff,0,strlen(buff));
+        len = tfp_sprintf(buff, "THR LIMIT %s", lookupTableThrottleLimitType[currentControlRateProfile->throttle_limit_type]);
+        if (currentControlRateProfile->throttle_limit_type != THROTTLE_LIMIT_TYPE_OFF) {
+            len = tfp_sprintf(buff, "%s %d", buff, currentControlRateProfile->throttle_limit_percent);
         }
+        displayWrite(osdDisplayPort, midCol - len/2, currentRow++, DISPLAYPORT_SEVERITY_NORMAL, buff);
 
+        specState = MOTOR;
```

Comment:
Another option is to use specState++; enum ordering will affect display order and adding elements would be slightly simpler.

Reply:
If somebody changes the order in the enum such a change would break. Good or bad depending on how you view it.